velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +325,86 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  int precision = getDecimalPrecisionScale(*fromType).first;
```

Comment:
auto [precision, scale] = getDecimalPrecisionScale(*fromType)

Reply:
This is to avoid a CI compilation error on macos. Under some compilation environment, below error is generated due to
> If a lambda-expression [...] captures a structured binding (explicitly or implicitly), the program is ill-formed.

```
/Users/distiller/project/./velox/expression/CastExpr-inl.h:384:43: error: reference to local binding 'scale' declared in enclosing function 'facebook::velox::exec::CastExpr::applyDecimalToVarcharCast'
          for (size_t i = 0; i < std::max(scale - countDigits(fraction), 0);
                                          ^
/Users/distiller/project/./velox/expression/CastExpr-inl.h:338:20: note: 'scale' declared here
  auto [precision, scale] = getDecimalPrecisionScale(*fromType);
```

Reply:
Can you try auto fromPrecisionScale = getDecimalPrecisionScale(*fromType); and use fromPrecisionScale.first and fromPrecisionScale.second to get the precision and scale?

Reply:
That should work, but looks to me currect way is clearer to understand. Maybe we can wait for the community suggestions.